aea/package_manager/v1.py

```diff
@@ -243,6 +243,12 @@ def verify(
                     continue
 
                 if calculated_hash != expected_hash:
+                    # this update is required for further dependency checks for
+                    # packages where this package is a dependency
+                    if self.is_third_party_package(package_id=package_id):
+                        self._third_party_packages[package_id] = calculated_hash
+                    if self.is_dev_package(package_id=package_id):
+                        self._dev_packages[package_id] = calculated_hash
```

Comment:
this is acceptable because it does not update any state in a way that leads to failure alternative behaviour the second time it is ran